packages/docsearch-react/package.json

```diff
@@ -42,5 +42,16 @@
     "@types/react": ">= 16.8.0 < 19.0.0",
     "react": ">= 16.8.0 < 19.0.0",
     "react-dom": ">= 16.8.0 < 19.0.0"
+  },
+  "peerDependenciesMeta": {
+    "@types/react": {
+      "optional": true
+    },
+    "react": {
+      "optional": true
+    },
+    "react-dom": {
+      "optional": true
+    }
```

Comment:
Have you tried applying this change at the `@docsearch/js` package level? I believe it wouldn't work but I'd rather not impact this one with unrelated changes.

Ultimately, I believe it's worth the workaround to prevent warnings on non-React packages

Reply:
Yeah it won't work. `peerDependenciesMeta` can define things "related to the dependencies listed in the `peerDependencies` field". So this workaround needs to be there in the package where we have issue causing peer deps.

Reply:
Also as you wrote at #1416, we need to make `@algolia/client-search` optional here: https://github.com/algolia/autocomplete/blob/next/packages/autocomplete-preset-algolia/package.json to fix all deps warnings (and work without error on Yarn v2/v3 and PNPM).

Reply:
> So this workaround needs to be there in the package where we have issue causing peer deps.

I think the workaround is worth it

> Also as you wrote at #1416, we need to make `@algolia/client-search` optional here: https://github.com/algolia/autocomplete/blob/next/packages/autocomplete-preset-algolia/package.json to fix all deps warnings (and work without error on Yarn v2/v3 and PNPM).

Indeed, but actually the issue is located at the Algolia JavaScript API client level (https://github.com/algolia/algoliasearch-client-javascript/issues/1080) not re-exporting types, so Autocomplete shouldn't fix it. We've fixed it in the next major of the API client along with many other changes.